### PR TITLE
feat: add cli commands to add/delete sourceNamespaces from AppProject

### PR DESCRIPTION
**Pull request by raghavi101**
The PR involves adding two commands for the 'Apps in Any Namespace' feature (#9755) :

`argocd proj add-source-namespace` : Enables users to add source namespaces in AppProject.Spec.SourceNamespaces, thereby whitelisting specific namespaces for the applications to be created in.
 ~`argocd proj delete-source-namespace`~  `argocd proj remove-source-namespace`: Allows users to remove source namespaces in AppProject.Spec.SourceNamespaces, thereby blacklisting specific namespaces.

Command usage :

- `argocd proj add-source-namespace PROJECT NAMESPACE`
 
Screenshot 1
Screenshot 2

-  `argocd proj delete-source-namespace PROJECT NAMESPACE` 
Screenshot 3

Screenshot 4

Checklist:
* [ ] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [x] The title of the PR conforms to the [Toolchain Guide](https://argo-cd.readthedocs.io/en/latest/developer-guide/toolchain-guide/#title-of-the-pr)
* [ ] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [x] Does this PR require documentation updates?
* [x] I've updated documentation as required by this PR.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md#legal)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).
* [ ] My new feature complies with the [feature status](https://github.com/argoproj/argoproj/blob/master/community/feature-status.md) guidelines.
* [x] I have added a brief description of why this PR is necessary and/or what this PR solves.
* [x] Optional. My organization is added to USERS.md.
* [ ] Optional. For bug fixes, I've indicated what older releases this fix should be cherry-picked into (this may or may not happen depending on risk/complexity).

### CODE REVIEW

File: cmd/argocd/commands/project.go [hunk 1]
```diff
@@ -509,6 +511,88 @@ func NewProjectAddSourceCommand(clientOpts *argocdclient.ClientOptions) *cobra.C
...
+			defer argoio.Close(conn)
+
+			proj, err := projIf.Get(ctx, &projectpkg.ProjectQuery{Name: projName})
+			errors.CheckError(err)
+
+			for _, item := range proj.Spec.SourceNamespaces {
+				if item == "*" && item == srcNamespace {
+					fmt.Printf("Source namespace '*' already allowed in project\n")
+					return
+				}
```
Comment (ishitasequeira) [inline cmd/argocd/commands/project.go:542]: I think the condition here needs to be like below. WDYT?

```
if (item == "*" || item == srcNamespace) {
    fmt.Printf("Source namespace %s already allowed in project\n", srcNamespace)
    return
}
```

File: cmd/argocd/commands/project.go [hunk 2]
```diff
@@ -509,6 +511,88 @@ func NewProjectAddSourceCommand(clientOpts *argocdclient.ClientOptions) *cobra.C
...
+		},
+	}
+	return command
+}
+
+// NewProjectDeleteSourceNamespace returns a new instance of an `argocd proj delete-source-namespace` command
+func NewProjectDeleteSourceNamespace(clientOpts *argocdclient.ClientOptions) *cobra.Command {
+	var command = &cobra.Command{
+		Use:   "delete-source-namespace PROJECT NAMESPACE",
+		Short: "Delete project source namespace",
```
Comment (ishitasequeira) [inline cmd/argocd/commands/project.go:556]: nit: `Removes the source namespace from the AppProject`

File: cmd/argocd/commands/project.go [hunk 3]
```diff
@@ -509,6 +511,88 @@ func NewProjectAddSourceCommand(clientOpts *argocdclient.ClientOptions) *cobra.C
...
+			proj.Spec.SourceNamespaces = append(proj.Spec.SourceNamespaces, srcNamespace)
+			_, err = projIf.Update(ctx, &projectpkg.ProjectUpdateRequest{Project: proj})
+			errors.CheckError(err)
+		},
+	}
+	return command
+}
+
+// NewProjectDeleteSourceNamespace returns a new instance of an `argocd proj delete-source-namespace` command
+func NewProjectDeleteSourceNamespace(clientOpts *argocdclient.ClientOptions) *cobra.Command {
```
Comment (ishitasequeira) [inline cmd/argocd/commands/project.go:553]: Let's call this is `RemoveSourceNamespace` instead of `DeleteSourceNamespace`

**Review by ishitasequeira — COMMENTED** [review]
You would need to execute `make clidocsgen` to regenerate the cli docs

**Review by jannfis — APPROVED** [review]
LGTM

Nice, thank you!

**Pull request merged**